projects/dsp-ui/src/lib/viewer/values/text-value/text-value-as-xml/text-value-as-xml.component.spec.ts

```diff
@@ -0,0 +1,418 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { TextValueAsXMLComponent } from './text-value-as-xml.component';
+import { Component, DebugElement, forwardRef, Input, OnInit, ViewChild } from '@angular/core';
+import { MockResource, ReadTextValueAsXml, UpdateTextValueAsXml } from '@dasch-swiss/dsp-js';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR, ReactiveFormsModule } from '@angular/forms';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { By } from '@angular/platform-browser';
+import { MatInputModule } from '@angular/material/input';
+import { CreateTextValueAsXml } from '@dasch-swiss/dsp-js/index';
+import { AppInitService } from '../../../../core';
+
+/**
+ * Test host component to simulate parent component.
+ */
+@Component({
+    selector: 'ckeditor',
+    template: ``,
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            multi: true,
+            useExisting: forwardRef(() => TestCKEditorComponent),
+        }
+    ]
+})
+class TestCKEditorComponent implements ControlValueAccessor {
+
+    @Input() config;
+
+    @Input() editor;
+
+    value;
+
+    onChange = (_: any) => {
+    };
+
+    constructor() {
+    }
+
+    writeValue(obj: any) {
+        this.value = obj;
+    }
+
+    registerOnChange(fn: any) {
+        this.onChange = fn;
+    }
+
+    registerOnTouched(fn: any) {
+    }
+
+    _handleInput(): void {
+        this.onChange(this.value);
+    }
+
+}
+
+/**
+ * Test host component to simulate parent component.
+ */
+@Component({
+    template: `
+        <dsp-text-value-as-xml #inputVal [displayValue]="displayInputVal" [mode]="mode"></dsp-text-value-as-xml>`
+})
+class TestHostDisplayValueComponent implements OnInit {
+
+    @ViewChild('inputVal') inputValueComponent: TextValueAsXMLComponent;
+
+    displayInputVal: ReadTextValueAsXml;
+
+    mode: 'read' | 'update' | 'create' | 'search';
+
+    ngOnInit() {
+
+        MockResource.getTestthing().subscribe(
+            res => {
+
+                this.displayInputVal = res.getValuesAs('http://0.0.0.0:3333/ontology/0001/anything/v2#hasRichtext', ReadTextValueAsXml)[0];
+
+                this.mode = 'read';
+            }
+        );
+
+    }
+}
+
+/**
+ * Test host component to simulate parent component.
+ */
+@Component({
+    template: `
+        <dsp-text-value-as-xml #inputVal [mode]="mode"></dsp-text-value-as-xml>`
+})
+class TestHostCreateValueComponent implements OnInit {
+
+    @ViewChild('inputVal') inputValueComponent: TextValueAsXMLComponent;
+
+    mode: 'read' | 'update' | 'create' | 'search';
+
+    ngOnInit() {
+
+        this.mode = 'create';
+
+    }
+}
+
+describe('TextValueAsXMLComponent', () => {
+
+    const appInitServiceMock = {
+        config: {
+            xmlTransform: {
+                "http://rdfh.ch/standoff/mappings/StandardMapping": {
```

Comment:
single quotes?

Reply:
done in https://github.com/dasch-swiss/dsp-ui-lib/pull/134/commits/2e7b3456557533e626891e138655faa23a444808